test/e2e/performanceprofile/functests/1_performance/cpu_management.go

```diff
@@ -887,8 +887,104 @@ var _ = Describe("[rfe_id:27363][performance] CPU Management", Ordered, func() {
 		})
 	})
 
+	Context("Check container runtimes cpu usage", func() {
+		var guaranteedPod, bestEffortPod *corev1.Pod
+		AfterEach(func() {
+			By("Cleaning up pods")
+			deleteTestPod(ctx, guaranteedPod)
+			deleteTestPod(ctx, bestEffortPod)
+		})
```

Comment:
you can avoid after each because the creation of the pods is done in the specific It(), you can convert that to [defer](https://go.dev/tour/flowcontrol/12) after each pods creation (as close as possible to the creation):

```
guaranteedPod = makePod(ctx, workerRTNode, true)
err := testclient.Client.Create(ctx, guaranteedPod)
Expect(err).ToNot(HaveOccurred(), "Failed to create guaranteed pod")
defer deleteTestPod(ctx, guaranteedPod)
```

Reply:
Revised with defer, Thanks for this suggestion.